### PR TITLE
provision: first implementation of the Digi Remote Manager provisioni…

### PR DESCRIPTION
**Pull request by diescalo**
…ng API

- Added a new API to allow customers to provision (register) devices in a Digi Remote Manager account individually or by batch. API is located in the 'DigiIoT.Maui.Utils.DRMUtils' class.
- Added the new class 'DRMAccount' that can be used by the new provisioning API or to provision a device or list of devices directly from the 'DRMAccount'.
- Exceptions generated by the new API are thrown as 'DRMException'. It can include an HTTP status code from the HTTP operation that failed.
- Added other model classes used to manage the requests and responses with DRM.

### CODE REVIEW

File: DigiIoT.Maui/Models/DRM/DRMAccount.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		/// Password of the Digi Remote Manager account.
+		/// </summary>
+		public string Password { get; }
+
+		/// <summary>
+		/// Class constructor. Instantiates a new <see cref="DRMAccount"/> object with the given 
+		/// parameters.
+		/// </summary>
+		/// <param name="username">Digi Remote Manager account username.</param>
+		/// <param name="password">Digi Remote Manager account password.</param>
```
Comment (rubenmoral) [inline DigiIoT.Maui/Models/DRM/DRMAccount.cs:43]: `/// <exception cref="ArgumentNullException">If the username or password are null.</exception>`

File: DigiIoT.Maui/Utils/DRMUtils.cs [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+		private const string ERROR_DEVICE_ID_EMPTY = "Device ID cannot be null or empty.";
+		private const string ERROR_DEVICE_ID_INVALID = "Device ID format is not valid. Expected format is 00000000-00000000-00000000-00000000.";
+		private const string ERROR_DEVICE_LIST_EMPTY = "Device list cannot be null or empty.";
+		private const string ERROR_PROVISIONING_FAILED = "Device provisioning failed. Status Code: {0}, Details: {1}";
+		private const string ERROR_NETWORK = "A network error occurred while provisioning the device.";
+		private const string ERROR_OTHER = "An unexpected error occurred.";
+
+		/// <summary>
+		/// Provisions a device to Digi Remote Manager using a <see cref="DRMAccount"/> and an optional install code.
```
Comment (rubenmoral) [inline DigiIoT.Maui/Utils/DRMUtils.cs]: `Provisions a single device...` for consistency with the next method?

**Review by rubenmoral — APPROVED** [review]

File: DigiIoT.Maui/Models/DRM/DRMAccount.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+		/// <exception cref="ArgumentNullException">If device ID is null.</exception>
+		/// <exception cref="ArgumentException">If the device ID is invalid.</exception>
+		/// <exception cref="DRMException">If the provisioning process fails.</exception>
+		public async Task<DeviceProvisionResult> ProvisionDevice(string deviceID, string installCode = null)
+		{
+			return await DRMUtils.ProvisionDevice(this, deviceID, installCode);
+		}
+
+		/// <summary>
+		/// Provisions a device to Digi Remote Manager using an optional install code.
```
Comment (daescalona) [inline DigiIoT.Maui/Models/DRM/DRMAccount.cs]: /// Provisions a list of devices to Digi Remote Manager.

File: DigiIoT.Maui/Types/DeviceProvisionTypes.cs [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+using Newtonsoft.Json;
+
+namespace DigiIoT.Maui.Types
+{
+
+    /// <summary>
+    /// Represents the response from Digi Remote Manager when provisioning multiple devices.
+    /// </summary>
+    internal class DeviceProvisionTypes
```
Comment (daescalona) [inline DigiIoT.Maui/Types/DeviceProvisionTypes.cs]: I think this name is missleading... Shouldn't it be `DeviceProvisionList` or something similar?

File: DigiIoT.Maui/Utils/DRMUtils.cs [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+		/// </summary>
+		/// <param name="drmAccount">The <see cref="DRMAccount"/> containing authentication credentials.</param>
+		/// <param name="devices">The list of devices to provision, each containing an ID and an optional install code.</param>
+		/// <returns>A task that returns a list of provisioning results for each device.</returns>
+		/// <exception cref="ArgumentNullException">If the DRM account or device list is null.</exception>
+		/// <exception cref="ArgumentException">If the device list is empty.</exception>
+		/// <exception cref="DRMException">If the request fails for all devices or the server response is invalid.</exception>
+		public static async Task<List<DeviceProvisionResult>> ProvisionDevices(DRMAccount drmAccount, List<DeviceProvisionRequest> devices)
+		{
+			if (drmAccount == null)
```
Comment (daescalona) [inline DigiIoT.Maui/Utils/DRMUtils.cs:100]: Check if list is null.

**Review by daescalona — APPROVED** [review]

File: DigiIoT.Maui/Utils/DRMUtils.cs (hunk 5 above)
Comment (diescalo) [reply DigiIoT.Maui/Utils/DRMUtils.cs:100]: Not necessary, it is checked in the 'ProvisionDevices' method.

**Pull request merged**